keps/sig-apps/0003-kubernetes-application-api.md

```diff
@@ -0,0 +1,1108 @@
+annotations, but, as each tool uses a different scheme, an application created by one tool is generally not recognizable 
+by another, and UI designers have no standard way to recognize and display an application created by any tool.
+
+Moreover, their is no way to determine if the application is healthy. Even if a UI is able to determine the components 
+and dependencies of the application, and even if it is able to determine that the individual components are healthy 
+(e.g All Services are created and all Deployments are fully replicated), this does not necessarily imply that the 
+application as a whole is able to service requests.
+
+### Goals
+
+1. Provide a standard API for creating, viewing, and managing applications in Kubernetes.
+1. Provide a CLI implementation, via kubectl, that interacts with the Application API.
+1. Provide installation status and garbage collection for applications.
+1. Provide a standard way for applications to surface a basic health check to the UIs.
+1. Provide an explicit mechanism for applications to declare dependencies on another application.
+1. Promote interoperability among ecosystem tools and UIs by creating a standard that tools MAY implement.
+
+### Non-Goals
+
+1. Create a standard that all tools MUST implement. 
+1. Provide a way for UIs to surface metrics from an application.
+1. Provide a method for the arbitrary aggregation of Kubernetes objects that do not comprise an Application.
+
+## Proposal
+In order to address the issues discussed in the [previous section](#motivation) we will introduce the Application Kind 
+to as a CRD extension the Kubernetes API, add the Application Controller to the Controller Manager, and modify 
+kubectl to make use the of the Application Kind.
+
+### User Stories
+
+#### Application Creation
+As an application maintainer or tool developer, I can create an application that is interoperable with other tools and 
+has first class support in Kuberentes UIs.
+
+#### Application View
+As an application user or UI designer, I have a standard way to view the components and status of Kubernetes 
+applications.
+
+#### Application Health
+As an application developer, I can surface a basic health check for UIs to consume, and, as a UI developer, I have a 
+standard way to consume application level health checks.
+
+#### Application Update
+As an application developer, maintainer, or administrator, I can update the configuration of my application.
+
+#### Application List
+As an application developer, maintainer or administrator, I can list the applications that are currently installed.
+
+#### Application Watch
+As an application developer, maintainer or administrator, I can watch the applications in a cluster for changes.
+
+#### Application Deletion
+As a user, I can delete my application, and when I delete my Application, its corresponding components are deleted.
+
```

Comment:
@kow3ns what about application rollback, marking an application as completed rollout or succesfuflly deployed. That is an important part of determining if this update was sucessfull or has issues and must be rolled back ?